dids/didjwk.go

```diff
@@ -0,0 +1,117 @@
+package dids
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/tbd54566975/web5-go/crypto"
+	"github.com/tbd54566975/web5-go/crypto/dsa"
+	"github.com/tbd54566975/web5-go/jwk"
+)
+
+type newDIDJWKOptions struct {
+	keyManager  crypto.KeyManager
+	algorithmID string
+}
+
+type newDIDJWKOption func(o *newDIDJWKOptions)
+
+// KeyManager is an option that can be passed to NewDIDJWK to provide a KeyManager
+func KeyManager(k crypto.KeyManager) newDIDJWKOption {
+	return func(o *newDIDJWKOptions) {
+		o.keyManager = k
+	}
+}
+
+// AlgorithmID is an option that can be passed to NewDIDJWK to specify a specific
+// cryptographic algorithm to use to generate the private key
+func AlgorithmID(id string) newDIDJWKOption {
+	return func(o *newDIDJWKOptions) {
+		o.algorithmID = id
+	}
+}
+
+// NewDIDJWK can be used to generate a new `did:jwk`. `did:jwk` is useful in scenarios where:
+//   - Offline resolution is preferred
+//   - Key rotation is not required
+//   - Service endpoints are not necessary
+//
+// Spec: https://github.com/quartzjer/did-jwk/blob/main/spec.md
+func NewDIDJWK(opts ...newDIDJWKOption) (BearerDID, error) {
```

Comment:
i've been thinking the same thing @amika-sq 

> Instead of returning a BearerDID, we return something else specific for the DHT DID method (e.g.: a DHTDID implementation). I know that golang doesn't support inheritance, so I'm not sure how that could relate back to a BearerDID. Perhaps we would need to fallback to an interface.

I tried this approach with `DIDJWK` by embedding `BearerDID` in a `DIDJWK` struct which technically will get the job done. people will just have to pass around `did.BearerDID` which i thought was a bit awkward.

If we keep things as is, the way to support more complex DID methods would be using plain ole functions e.g. `UpdateDIDDHT(did *dids.BearerDID)`, `DeactivateDIDDHT(did *dids.BearerDID)`